### PR TITLE
Fix agent healthchecks to accept any HTTP 2xx response as healthy

### PR DESCRIPTION
**Pull request by jakolehm**
Fixes partially #1790

### CODE REVIEW

File: agent/lib/kontena/workers/container_health_check_worker.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class ContainerHealthCheckWorker
     include Kontena::Helpers::PortHelper
     include Kontena::Helpers::RpcHelper
 
-    HEALTHY_STATUSES = [200]
+    HEALTHY_STATUSES = (200 .. 208).to_a.freeze
```
Comment (SpComb) [inline agent/lib/kontena/workers/container_health_check_worker.rb]: Should this just be `200...300` to also cover any custom codes up to 299?

The `to_a` should be unnecessary?

```
irb(main):005:0> (200...300).include? 206
=> true
```

**Comment by SpComb** [comment]
Accepting any 2xx status as healthy sounds reasonable to me from a design perspective. Complete fix for #1790 needs some kind of configuration/label, but this is a good first step IMO.

File: agent/lib/kontena/workers/container_health_check_worker.rb (hunk 1 above)
Comment (SpComb) [reply agent/lib/kontena/workers/container_health_check_worker.rb]: Should be `HEALTHY_STATUSES = (200...300)` to actually match any 2xx status codes, not just up to 208. There are [standardized](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success) HTTP response codes beyond 208.

**Comment by jakolehm** [comment]
@SpComb PTAL.

File: agent/lib/kontena/workers/container_health_check_worker.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class ContainerHealthCheckWorker
     include Kontena::Helpers::PortHelper
     include Kontena::Helpers::RpcHelper
 
-    HEALTHY_STATUSES = [200]
+    HEALTHY_STATUSES = (200 .. 299).to_a.freeze
```
Comment (SpComb) [inline agent/lib/kontena/workers/container_health_check_worker.rb]: Any particular reason to use `to_a` here? I just imagine that `Range#include?` is going to be more efficient than `Array#include?` with 100 elements, but it doesn't really matter.

**Pull request merged**